edisgo/flex_opt/check_tech_constraints.py

```diff
@@ -97,180 +95,240 @@ def lv_line_load(edisgo_obj):
+        :pandas:`pandas.Timestamp<Timestamp>`, and 'voltage_level' specifying
+        the voltage level the line is in (either 'mv' or 'lv').
 
     """
-    # get lines and nominal voltage
+    if edisgo_obj.results.i_res.empty:
+        raise Exception(
+            "No power flow results to check over-load for. Please perform "
+            "power flow analysis first."
+        )
+
+    # get lines in voltage level
     mv_grid = edisgo_obj.topology.mv_grid
     if voltage_level == "lv":
-        lines_df = edisgo_obj.topology.lines_df[
+        lines = edisgo_obj.topology.lines_df[
             ~edisgo_obj.topology.lines_df.index.isin(mv_grid.lines_df.index)
-        ]
-        lv_grids = list(edisgo_obj.topology.lv_grids)
-        if len(lv_grids) > 0:
-            nominal_voltage = lv_grids[0].nominal_voltage
-        else:
-            nominal_voltage = np.NaN
+        ].index
     elif voltage_level == "mv":
-        lines_df = mv_grid.lines_df
-        nominal_voltage = mv_grid.nominal_voltage
+        lines = mv_grid.lines_df.index
     else:
         raise ValueError(
-            "{} is not a valid option for input variable 'voltage_level' in "
-            "function lines_allowed_load. Try 'mv' or "
-            "'lv'.".format(voltage_level)
+            "{} is not a valid option for input variable 'voltage_level'. "
+            "Try 'mv' or 'lv'.".format(voltage_level)
         )
 
-    i_lines_allowed_per_case = {}
-    i_lines_allowed_per_case["feed-in_case"] = (
-        lines_df.s_nom
-        / sqrt(3)
-        / nominal_voltage
-        * edisgo_obj.config["grid_expansion_load_factors"][
-            "{}_feed-in_case_line".format(voltage_level)
-        ]
-    )
-
-    # adapt i_lines_allowed for radial feeders
-    buses_in_cycles = list(
-        set(itertools.chain.from_iterable(edisgo_obj.topology.rings))
-    )
-
-    # Find lines in cycles
-    lines_in_cycles = list(
-        lines_df.loc[
-            lines_df[["bus0", "bus1"]].isin(buses_in_cycles).all(axis=1)
-        ].index.values
-    )
-    lines_radial_feeders = list(
-        lines_df.loc[~lines_df.index.isin(lines_in_cycles)].index.values
-    )
-
-    # lines in cycles have to be n-1 secure
-    i_lines_allowed_per_case["load_case"] = (
-        lines_df.loc[lines_in_cycles].s_nom
-        / sqrt(3)
-        / nominal_voltage
-        * edisgo_obj.config["grid_expansion_load_factors"][
-            "{}_load_case_line".format(voltage_level)
-        ]
-    )
+    # calculate relative line load and keep maximum over-load of each line
+    relative_i_res = lines_relative_load(edisgo_obj, lines)
 
-    # lines in radial feeders are not n-1 secure anyways
-    i_lines_allowed_per_case["load_case"] = pd.concat(
-        [
-            i_lines_allowed_per_case["load_case"],
-            lines_df.loc[lines_radial_feeders].s_nom / sqrt(3) / nominal_voltage,
-        ]
-    )
+    crit_lines_relative_load = relative_i_res[relative_i_res > 1].max().dropna()
```

Comment:
I think it is correct like this because the relative loading is the ratio of allowed loading and actual loading. The allowed loading can be increased by changing the load factor in the config to e.g. 1.1 instead of 1. But the relative loading should never exceed 1.